addons/SpellBook/SpellBook.lua

```diff
@@ -0,0 +1,382 @@
+_addon.name     = 'SpellBook'
+_addon.author   = 'SigilBaram'
+_addon.version  = '0.0.1'
+_addon.commands  = {'spellbook','spbk'}
+
+require('tables')
+res = require('resources')
+
+spell_types = {
+    whitemagic= { type = 'WhiteMagic', readable = 'White Magic spells' },
+    blackmagic= { type = 'BlackMagic', readable = 'Black Magic spells' },
+    songs=      { type = 'BardSong', readable = 'Bard songs' },
+    ninjutsu=   { type = 'Ninjutsu', readable = 'Ninjutsu' },
+    summoning=  { type = 'SummonerPact', readable = 'Summoning spells' },
+    bluemagic=  { type = 'BlueMagic',  readable = 'Blue Magic spells' },
+    geomancy=   { type = 'Geomancy', readable = 'Geomancy spells' },
+    trusts=     { type = 'Trust', readable = 'Trusts'},
+    all=        { type = 'all', readable = 'spells of all types'}
+}
+
+windower.register_event('addon command', function (command, ...)
+    local args = L{...}
+    local player = windower.ffxi.get_player()
```

Comment:
It's not commonly known, but this function (`windower.ffxi.get_player`) is pretty expensive. You use it once here and then again several times later, including in every `is_learnable` call, which is a lot. So much that it could lag out older PCs all on its own. Two ways around that, either pass that variable you get here to the nested function calls, or create a global variable for the player and update it on every login/logout.